fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1374,6 +1375,40 @@ void checkFieldInfoCountingAfterMerge() {
         assertTrue(mergeHappened);
     }
 
+    @Test
+    void testMultipleUpdateSegments() {
+        final Index index = COMPLEX_GROUPED_WITH_PRIMARY_KEY_SEGMENT_INDEX;
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER, 3.0)
+                .build();
+        try (FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            for (int i = 0; i < 20; i++) {
+                recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(i, "", "", 0));
+            }
+            context.commit();
+        }
+        final long segmentCountBefore;
+        try (FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            segmentCountBefore = getSegmentCount(index, Tuple.from(0));
+        }
+        try (FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            for (int i = 0; i < 10; i++) {
+                recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(i, numbersText(i), "", 0));
+            }
+            context.commit();
+        }
+        assertThat(timer.getCount(LuceneEvents.Events.LUCENE_DELETE_DOCUMENT_BY_QUERY), lessThan(5));
```

Comment:
Debugging this test against main, it looks like `LUCENE_FIND_MERGES` is 14, would it make sense to assert about that counter too? Locally it appears to be 4, which still seems higher than I would expect, but making sure it's less than 9 seems good. Of course that wouldn't apply if we defer merges. Deferring merges would also effect the segment count assertions.